BUG: Tackle GH47203; Ensure body is non-empty when accessing row_body_cells

tomaarsen opened the pull request:
- [x] closes #47203
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests).
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v1.5.0.rst` file.

tomaarsen committed:
BUG: Tackle GH47203; Ensure body is non-empty when accessing row_body_cells

tomaarsen committed:
BUG: Tackle GH47203; Add simple test case with empty DataFrames

I followed https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#test-structure which seemed to prefer simple functions as tests, rather than embedding them in classes. I can easily change this up if requested.

tomaarsen committed:
BUG: Tackle GH47203; Add whatsnew entry

tomaarsen committed:
Moved tests to `io/formats/style/test_to_latex.py`

from `io/formats/test_to_latex.py`

mroeschke commented on pandas/tests/io/formats/style/test_to_latex.py:
```diff
@@ -1032,3 +1032,15 @@ def test_concat_recursion():
+)
+def test_empty_clines(df: DataFrame):
+    # GH 47203
+    df.style.to_latex(clines="all;data")
```
Could you assert the latex result of this call?

tomaarsen replied:
Done and dusted! Good call. I've also added a simple parametrize over all the legal `clines` arguments. Should be all good to go.

tomaarsen committed:
BUG: GH47203; Add expected return to test cases

And parametrize over clines as well. 'skip-last;data' was also broken, after all.

jreback commented:
thanks @tomaarsen
